src/core/controller/index.ts

```diff
@@ -238,8 +256,9 @@ export class Controller {
 	async togglePlanActModeWithChatSettings(chatSettings: ChatSettings, chatContent?: ChatContent): Promise<boolean> {
 		const didSwitchToActMode = chatSettings.mode === "act"
 
-		// Store mode in-memory only
+		// Store mode in-memory and persist to global state
 		this.mode = chatSettings.mode
+		await updateGlobalState(this.context, "mode", this.mode)
```

Comment:
Evan claims its better to have this as a workspace setting because otherwise it might interfere with other workspaces and their modes.